monitor/dockermonitor/docker.go

```diff
@@ -469,22 +476,24 @@ func (d *dockerMonitor) handleStartEvent(event *events.Message) error {
 	info, err := d.dockerClient.ContainerInspect(context.Background(), dockerID)
 
 	if err != nil {
-		//If we see errors, we will kill the container for security reasons.
-		if err := d.dockerClient.ContainerStop(context.Background(), dockerID, &timeout); err != nil {
-			log.WithFields(log.Fields{
-				"package": "monitor",
-				"error":   err.Error(),
-			}).Warn("Failed to stop illegal container")
-		}
-
-		d.collector.CollectContainerEvent(&collector.ContainerRecord{
-			ContextID: contextID,
-			IPAddress: "N/A",
-			Tags:      nil,
-			Event:     collector.ContainerFailed,
-		})
+		// If we see errors, we will kill the container for security reasons if DockerMonitor was configured to do so.
+		if d.killContainerError {
+			if err := d.dockerClient.ContainerStop(context.Background(), dockerID, &timeout); err != nil {
+				log.WithFields(log.Fields{
+					"package": "monitor",
+					"error":   err.Error(),
+				}).Warn("Failed to stop illegal container")
+			}
 
-		return fmt.Errorf("Cannot read container information. Killing container. ")
+			d.collector.CollectContainerEvent(&collector.ContainerRecord{
+				ContextID: contextID,
+				IPAddress: "N/A",
+				Tags:      nil,
+				Event:     collector.ContainerFailed,
+			})
+			return fmt.Errorf("Cannot read container information. Killing container. ")
+		}
+		return fmt.Errorf("Cannot read container information. Container still alive per policy. ")
```

Comment:
Same as above why are we returning an error. If we keep container around, how would we deal with this container later ? whats the expectation ?